File: go/485_Max_Consecutive_Ones.go
func findMaxConsecutiveOnes(nums []int) int {
    var maxCount int
    var tempCount int
        
    for _, num := range nums {
        if (num == 0) {
            maxCount = max(maxCount, tempCount);
            tempCount = 0
        } else {
            tempCount++
        }
    }
    maxCount = max(maxCount, tempCount);
    
    return maxCount
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}

File: kotlin/485_Max_Consecutive_Ones.kt
import kotlin.math.max

class Solution {
    fun findMaxConsecutiveOnes(nums: IntArray): Int {
        var maxCount = 0;
        var tempCount = 0;
            
        for (num in nums) {
            if (num == 0) {
                maxCount = max(maxCount, tempCount);
                tempCount = 0;
            } else {
                tempCount++;
            }
        }
        maxCount = max(maxCount, tempCount);
        
        return maxCount;
    }
}
